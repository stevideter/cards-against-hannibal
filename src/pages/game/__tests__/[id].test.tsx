jest.mock('next/router');
jest.mock('swr');
jest.mock('../../../components/GameBoard/GameBoard', () => 'GameBoard');

import Game from '../[id]';
import renderer from 'react-test-renderer';
import { useRouter } from 'next/router';
import useSWR from 'swr';

describe('Game', () => {
    it('renders loading on stale', () => {
        (useRouter as jest.Mock<any>).mockReturnValue({
            query: {
                id: 'id',
            },
        });
        (useSWR as jest.Mock<any>).mockReturnValue({});
        const tree = renderer.create(<Game nickname="nick" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders game on data', () => {
        (useRouter as jest.Mock<any>).mockReturnValue({
            query: {
                id: 'id',
            },
        });
        const data = {
            players: [
                {
                    hand: [],
                },
            ],
            blackCards: [],
            whiteCards: [],
        };
        (useSWR as jest.Mock<any>).mockReturnValue({
            data,
        });
        const tree = renderer.create(<Game nickname="nick" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders error on swr error', () => {
        (useRouter as jest.Mock<any>).mockReturnValue({
            query: {
                id: 'id',
            },
        });
        const error = {
            message: 'error',
        };
        (useSWR as jest.Mock<any>).mockReturnValue({
            error,
        });
        const tree = renderer.create(<Game nickname="nick" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
