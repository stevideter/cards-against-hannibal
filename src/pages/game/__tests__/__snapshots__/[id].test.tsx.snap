// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders error on swr error 1`] = `
<div>
  Failed to load
</div>
`;

exports[`Game renders game on data 1`] = `
<div>
  <p>
    Game: 
    id
  </p>
  <p>
    Player: 
    nick
  </p>
  <div>
    Players
  </div>
  <div>
    <div>
      name
    </div>
  </div>
  <GameBoard
    blackCards={Array []}
    whiteCards={Array []}
  />
</div>
`;

exports[`Game renders loading on stale 1`] = `
<div>
  loading...
</div>
`;
